oxidation/crates/poc/enum.json

```diff
@@ -0,0 +1,29 @@
+{
+    "name": "FooEnum",
+    "variants": [
+        {
+            "name": "Foo0",
+            "fields": [
+                {
+                    "name": "Bar0",
+                    "type": "String",
+                    "rename": "bar0"
+                },
+                {
+                    "name": "Bar1",
+                    "type": "Vec<u8>",
```

Comment:
Type should be not related to Rust (we have another an implementation in Python after all ^^)

So the types are:
 - generics: `Map` `List` `Enum` `Nested`
 - scalars: `Boolean` `Integer` `Float` `String` `Bytes` 

We may want to add additional scalars in the future (e.g. non-zero integer, non-negative integers etc.) but it's easy to add them when we will need them

Reply:
And as we say those types must be deserialized to propor structs:

```
enum AnyType {
  Scalar(ScalarType),
 ...
}

struct ScalarType {
  name: String
}

impl TryFrom<syn::Type> for ScalarType {
  ...
}

struct MapType {
  item_type: AnyType,
}

impl TryFrom<syn::Type> for MapType {
  ...
}

[...]
```